application/browser/application_security_policy.cc

```diff
@@ -76,9 +76,9 @@ bool ApplicationSecurityPolicy::IsAccessAllowed(const GURL& url) const {
   if (!enabled_)
     return true;
 
-  // Accessing own resources is always allowed.
-  if (url.SchemeIs(application::kApplicationScheme) &&
-      url.host() == app_->id())
+  // Accessing own resources in W3C Widget is always allowed.
+  if (app_->data()->manifest_type() == Manifest::TYPE_WIDGET &&
```

Comment:
After enable 'scope' field, the old check is not enough to XPK package in some conditions, it has to check the url 'path' if there's a 'scope' set in manifest.json.